src/controllers/project.js

```diff
@@ -6,6 +6,36 @@ const User = require('../models/user');
 const isApplicableAmount = (amount, collected, requested) =>
   requested <= amount - collected;
 
+
+const filterProjects = async (req, res) => {
+  try {
+    const { category } = req.query;
+    const projects = await Project.find({ 'category.content': category });
+    return res.json(projects);
+  } catch (error) {
+    return res.status(400).json(error);
+  }
+};
+
+const searchProjects = async (req, res) => {
+  try {
+    const { search } = req.query;
+    // the user do not necessarily search for a project by its category
```

Comment:
maybe now we can remove these comments

Reply:
ok comments deleted
and I made some new changes to the query because I added the categories to the project model
so I had to make few changes
as for the tests will add them asap